src/NLU.DevOps.Dialogflow/DialogflowNLUTestClient.cs

```diff
@@ -74,9 +74,12 @@ protected override async Task<LabeledUtterance> TestAsync(string utterance, Canc
                         var client = await this.GetSessionClientAsync(cancellationToken).ConfigureAwait(false);
                         var result = await client.DetectIntentAsync(sessionName, queryInput, cancellationToken).ConfigureAwait(false);
                         return new LabeledUtterance(
-                            result.QueryResult.QueryText,
-                            result.QueryResult.Intent.DisplayName,
-                            result.QueryResult.Parameters?.Fields.SelectMany(GetEntities).ToList());
+                                result.QueryResult.QueryText,
+                                result.QueryResult.Intent.DisplayName,
+                                result.QueryResult.Parameters?.Fields.SelectMany(GetEntities).ToList())
+                            .WithScore(result.QueryResult.IntentDetectionConfidence)
+                            .WithTextScore(result.QueryResult.SpeechRecognitionConfidence)
+                            .WithTimestamp(DateTimeOffset.Now);
```

Comment:
Should we bubble the `WithTimestamp` up to a different abstraction level (i.e., so each service doesn't have to do this individually)?

Reply:
Maybe in a future PR.